File: leetcode/easy/70-climb-stairs/iterative.cpp
/**
 * @file iterative.cpp
 * @author Vortexx2
 * @brief Problem 70
 * @date 31-07-2021
 * 
 * Runtime - 0 ms
 * Memory Usage - 5.9 MB
 */



class Solution
{
public:
  int climbStairs(int n)
  {
    if (n <= 3)
      return n;
    int a = 1, b = 2;
    int res;

    for (int i = 2; i < n; i++)
    {
      res = a + b;
      a = b;
      b = res;
    }
  }
};

File: leetcode/easy/70-climb-stairs/iterative.py
"""
  @author Vortexx2
  Problem 70 - Climbing Stairs

  Runtime - 43 ms
  Memory Usage - 13.8 MB
"""
class Solution:
  def climbStairs(self, n: int) -> int:

    if (n <= 3):
      return n

    a, b, res = 1, 2, 0

    for _ in range(n - 2):
      res = a + b
      a = b
      b = res

    return res
